src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -11715,6 +11737,179 @@ void Test1(object x1, object? y1)
                 );
         }
 
+        [Fact]
+        public void ConditionalBranching_14()
+        {
+            var compilation = CreateCompilation(@"
+class C
+{
+    C? _cField = null;
+    C _nonNullCField = new C();
+    C? GetC() => null;
+    C? CProperty { get => null; }
+
+    void Test1(C? c1)
+    {
+        if (c1?._cField != null)
+        {
+            c1._cField.ToString();
+        }
+        else
+        {
+            c1._cField.ToString(); // warn
+        }
+    }
+
+    void Test2()
+    {
+        C? c2 = GetC();
+        if (c2?._cField != null)
+        {
+            c2._cField.ToString();
+        }
+        else
+        {
+            c2._cField.ToString(); // warn x2
+        }
+    }
+
+    void Test3(C? c3)
+    {
+        if (c3?._cField?._cField != null)
+        {
+            c3._cField._cField.ToString();
+        }
+        else if (c3?._cField != null)
+        {
+            c3._cField.ToString();
+            c3._cField._cField.ToString(); // warn
```

Comment:
Nit: to maximize readability, I suggest using unique labels `// warn 1`, `// warn 2`, etc so it's really easy to match the diagnostics to the code.